service/routes.py

```diff
@@ -294,288 +318,192 @@ def post(self):
       )
 
     app.logger.info('Payload = %s', api.payload)
+    data = api.payload
+
+    if not isinstance(data, dict):
+      abort(status.HTTP_400_BAD_REQUEST, "Expected a json request body")
+
     wishlist = Wishlist()
-    wishlist.deserialize(api.payload)
+    wishlist.deserialize(data)
     wishlist.create()
     data = wishlist.serialize()
 
     location_url = api.url_for(WishlistResource, wishlist_id=wishlist.id, _external=True)
     return data, status.HTTP_201_CREATED, {'Location': location_url}
 
 ######################################################################
-#  PATH: /wishlists/{wishlist_id}/products/{product_id}
+#  PATH: /wishlists/{wishlist_id}/products
 ######################################################################
-# @api.route('/wishlists/<wishlist_id>/products/<product_id>')
-# @api.param('wishlist_id', 'The Wishlist identifier')
-# @api.param('product_id', 'The Product identifier')
-class ProductResource(Resource):
+@api.route('/wishlists/<wishlist_id>/products')
+@api.param('wishlist_id', 'The Wishlist identifier')
+class ProductCollectionResource(Resource):
   """
-  ProductResource class
+  ProductCollectionResource class
 
-  Allows the manipulation on a a single Product in a Wishlist
-  GET - Returns the product in the wishlist
-  POST - Create the product in the wishlist
-  DELETE - Delete the product from the wishlist
+  Allows the manipulation on a collection of Products in a Wishlist
+  POST - Create a product in a wishlist
   """
+  @api.doc('create_a_product')
+  @api.response(415,"Unsupported media type : application/json expected")
+  @api.response(400,"Expected a json request body")
+  @api.expect(create_product_model)
+  def post(self,wishlist_id):
```

Comment:
I think this function will return a code ```201``` and the created product.

Reply:
solved